packages/contract/src/DaimoPaymaster.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.12;
+
+import "account-abstraction/core/BasePaymaster.sol";
+
+import "./DaimoAccount.sol";
+
+contract DaimoPaymaster is BasePaymaster {
+    mapping(address => bool) public destWhitelist;
+    mapping(address => bool) public senderBlacklist;
+
+    event UserOperationSponsored(
+        bytes32 indexed userOpHash,
+        uint256 requiredPreFund
+    );
+
+    constructor(
+        IEntryPoint _entryPoint,
+        address _owner
+    ) BasePaymaster(_entryPoint) {
+        transferOwnership(_owner);
+    }
+
+    function setDestAddressWhitelist(
+        address[] calldata addresses,
+        bool isWhitelisted
+    ) public onlyOwner {
+        for (uint256 i = 0; i < addresses.length; i++) {
+            destWhitelist[addresses[i]] = isWhitelisted;
+        }
+    }
+
+    function setSenderAddressBlacklist(
+        address[] calldata addresses,
+        bool isBlacklisted
+    ) public onlyOwner {
+        for (uint256 i = 0; i < addresses.length; i++) {
+            senderBlacklist[addresses[i]] = isBlacklisted;
+        }
+    }
+
+    function _validatePaymasterUserOp(
+        UserOperation calldata userOp,
+        bytes32 userOpHash,
+        uint256 requiredPreFund
+    ) internal override returns (bytes memory context, uint256 validationData) {
```

Comment:
so if i understand right, we sponsor ops that:
- call `executeBatch` selector on their own acccount
- each call in the batch is either a self-call or in the address whitelist

...but no way to tell whether the account is a Daimo account (ie, from the DaimoAccountFactory) or not, right?

not necessarily an issue--i dont think anyone will grief us, and our downside is limited if they do.

Reply:
i was thinking about adding a check whether an account is from DaimoAccountFactory, but since accounts we make our upgradeable in theory someone could just upgrade them to a different implementation to grief us, so it doesn't add much protection.